middleware.js

```diff
@@ -53,7 +53,19 @@ module.exports = function(compiler, options) {
 	if(typeof options.reporter !== "function") options.reporter = defaultReporter;
 
 	// store our files in memory
-	var fs = compiler.outputFileSystem = new MemoryFileSystem();
+	var fs;
+	if(options.fileSystem) {
```

Comment:
Does your usecase still work when you remove the `options.fileSystem`? It looks now as if you've implemented two ways of reusing the `outputFileSystem`; by allowing `options.fileSystem` AND by checking if `compiler.outputFileSystem` already exists.